src/app/modules/item/services/item-task-answer.service.ts

```diff
@@ -1,41 +1,47 @@
 import { Injectable, OnDestroy } from '@angular/core';
-import { combineLatest, EMPTY, forkJoin, interval, Observable, of, Subject } from 'rxjs';
+import { combineLatest, EMPTY, forkJoin, interval, merge, Observable, of, ReplaySubject, Subject } from 'rxjs';
 import {
   catchError,
   delayWhen,
   distinctUntilChanged,
   filter,
-  map,
   mapTo,
-  repeat,
   shareReplay,
+  startWith,
   switchMap,
   switchMapTo,
   take,
   takeUntil,
-  tap,
+  timeout,
   withLatestFrom,
 } from 'rxjs/operators';
 import { SECONDS } from 'src/app/shared/helpers/duration';
 import { errorIsHTTPForbidden } from 'src/app/shared/helpers/errors';
+import { isNotNull } from 'src/app/shared/helpers/null-undefined-predicates';
+import { repeatLatestWhen } from 'src/app/shared/helpers/repeatLatestWhen';
 import { AnswerTokenService } from '../http-services/answer-token.service';
 import { Answer, CurrentAnswerService } from '../http-services/current-answer.service';
 import { GetAnswerService } from '../http-services/get-answer.service';
 import { GradeService } from '../http-services/grade.service';
 import { ItemTaskInitService } from './item-task-init.service';
 
-const answerAndStateSaveInterval = 5*SECONDS;
+const answerAndStateSaveInterval = (): number => ((window as unknown as Record<string, number>).__DELAY ?? 5)*SECONDS;
```

Comment:
Yes it was supposed to be removed before merge, thus the ugliness :smile: 
If we want to add our own globals, we can extend the Window interface in a TS type file, it will avoid this typing horror

About the constant itself, where could we isolate it ?

Reply:
fixed